rs/nns/governance/src/neuron/types.rs

```diff
@@ -500,20 +523,28 @@ impl Neuron {
         if topic == Topic::ExchangeRate {
             return;
         }
+
+        // Data migration for updating the recent ballots so we can use a circular buffer here.
+        if self.recent_ballots_next_entry_index.is_none() {
+            self.recent_ballots.reverse();
+            self.recent_ballots_next_entry_index =
+                Some(self.recent_ballots.len() % MAX_NEURON_RECENT_BALLOTS);
+        }
+
         let ballot_info = BallotInfo {
             proposal_id: Some(*proposal_id),
             vote: vote as i32,
         };
-        // We would really like to have a circular buffer here. As
-        // we're dealing with a simple vector, we insert at the
-        // beginning and remove at the end once we have reached
-        // the maximum number of votes to keep track of.
-        self.recent_ballots.insert(0, ballot_info);
-        // Pop and discard elements from the end until we reach
-        // the maximum allowed length of the vector.
-        while self.recent_ballots.len() > MAX_NEURON_RECENT_BALLOTS {
-            self.recent_ballots.pop();
+
+        // Vector is full
+        if self.recent_ballots.len() >= MAX_NEURON_RECENT_BALLOTS {
+            self.recent_ballots[self.recent_ballots_next_entry_index.unwrap()] = ballot_info;
```

Comment:
To avoid unwrapping, consider refactoring the setting of `recent_ballots_next_entry_index` to a function `ensure_some_next_entry_index`, which sets the index as above and returns it as a `u32`. So you would have `ensure_some_next_entry_index(self.recent_ballots_next_entry_index)` instead of `self.recent_ballots_next_entry_index.unwrap()`.

Reply:
adjusted the migration code above to get the index so it's not unwrapped here.  Good suggestion, thanks.